scripts/ec2/ls.coffee

```diff
@@ -56,82 +63,106 @@ listEC2Instances = (params, complete, error) ->
   expiration_tag = instance.Tags.filter get_expiration_tag
   if expiration_tag.length == 0
     return false
 
-  is_not_expired_now = expiration_tag > moment()
+  expiraton_moment = moment(expiration_tag[0].Value).format('YYYY-MM-DD')
+
+  is_not_expired_now = expiraton_moment < moment().format('YYYY-MM-DD')
   return is_not_expired_now
 
+extract_message = (instances, msg)->
+  return msg + messages_from_ec2_instances(instances)
+
+get_instance_tag = (instance, key, default_value = "")->
+  tags = _.filter(instance.Tags, (tag)-> return tag.Key == key)
+  if not _.isEmpty(tags)
+    return tags[0].Value
+  else
+    return default_value
+
 handle_instances = (robot) ->
+  msg_room = (msg_text, room = process.env.HUBOT_EC2_MENTION_ROOM) ->
+    robot.messageRoom room, msg_text
+
   return (instances) ->
+    instances_that_expired = instances.filter instance_has_expired
     instances_that_will_expire = instances.filter instance_will_expire_soon
-    instances_that_have_expired = instances.filter instance_has_expired
 
-    robot.messageRoom process.env.HUBOT_EC2_MENTION_ROOM, "Instances that will expire soon...\n" + 
-      messages_from_ec2_instances(instances_that_will_expire)
+    msg_text_expired = extract_message(instances_that_expired, EXPIRED_MESSAGE)
+    msg_text_expire_soon = extract_message(instances_that_will_expire, EXPIRES_SOON_MESSAGE)
 
-    robot.messageRoom process.env.HUBOT_EC2_MENTION_ROOM, "Instances that have expired...\n" + 
-      messages_from_ec2_instances(instances_that_have_expired)
+    msg_room(msg_text_expired)
+    msg_room(msg_text_expire_soon)
 
-    instance_ids_to_stop = []
-    for instance in instances_that_have_expired
-      instance_ids_to_stop.push(instance.InstanceId)
+    for user in _.values(robot.brain.data.users)
+      creator_email = user.email_address || "_DL_CFPB_Software_Delivery_Team@cfpb.gov"
+      user_id = user.id || 1
 
-    ec2.stopInstances { InstanceIds: instance_ids_to_stop }, (err, res) ->
+      user_instances = _.filter(instances_that_will_expire, (this_instance)-> return get_instance_tag(this_instance, 'Creator') == creator_email)
+
+      if user_instances
+        msg_text_expire_soon = extract_message(user_instances, USER_EXPIRES_SOON_MESSAGE) + EXTEND_COMMAND
+
+        robot.send({user: user_id}, msg_text_expire_soon)
+
+    ec2.stopInstances {InstanceIds: _.pluck(instances_that_expired, 'InstanceId')}, (err, res) ->
       if err
-        robot.messageRoom process.env.HUBOT_EC2_MENTION_ROOM, err
+        msg_room(res)
+
 
 handle_ec2_instance = (robot) ->
   if process.env.HUBOT_EC2_MENTION_ROOM
-    params = {}
-    params['Filters'] = [{ Name: 'tag:CreatedByApplication', Values: ['chat'] }]
-    listEC2Instances(params, handle_instances(robot), ->)  
-    
+    params = getArgParams(null, filter = "chat")
+    listEC2Instances(params, handle_instances(robot), ->)
+
 ec2_setup_polling = (robot) ->
   setInterval ->
     handle_ec2_instance(robot)?
   , 1000 * 60 * 60 * 8
```

Comment:
Polling interval should be at nyquist frequency of the interval and on the hour configurable.